File: app/src/main/java/com/ryanpconnors/art_thief/rate/ArtWorkListViewHolder.java
package com.ryanpconnors.art_thief.rate;

import android.content.Context;
import android.graphics.Bitmap;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.ryanpconnors.art_thief.R;
import com.ryanpconnors.art_thief.artgallery.ArtWork;
import com.ryanpconnors.art_thief.artgallery.Gallery;
import com.ryanpconnors.art_thief.rate.ArtWorkListFragment.OnArtWorkListFragmentInteractionListener;

import java.util.Locale;

/**
 * Created by Ryan Connors on 2/17/16.
 */
public class ArtWorkListViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    private ImageView mArtWorkImageView;
    private Button mArtworkTakenButton;
    private TextView mShowIdTextView;
    private TextView mTitleTextView;
    private TextView mArtistTextView;
    private TextView mMediaTextView;
    private ImageView mArtWorkRatingStarImageView;
    private TextView mArtWorkRatingTextView;

    private OnArtWorkListFragmentInteractionListener mListener;

    private ArtWork mArtWork;

    public ArtWorkListViewHolder(View itemView) {
        super(itemView);
        itemView.setOnClickListener(this);

        mArtWorkImageView = (ImageView) itemView.findViewById(R.id.artwork_list_item_artwork_image);
        mArtworkTakenButton = (Button) itemView.findViewById(R.id.artwork_list_item_taken_button);
        mShowIdTextView = (TextView) itemView.findViewById(R.id.artwork_list_item_show_id);
        mTitleTextView = (TextView) itemView.findViewById(R.id.artwork_list_item_title);
        mArtistTextView = (TextView) itemView.findViewById(R.id.artwork_list_item_artist);
        mMediaTextView = (TextView) itemView.findViewById(R.id.artwork_list_item_media);
        mArtWorkRatingStarImageView = (ImageView) itemView.findViewById(R.id.artwork_list_item_rating_star_image);
        mArtWorkRatingTextView = (TextView) itemView.findViewById(R.id.artwork_list_item_rating_text_view);
    }

    public void bindArtWork(ArtWork artWork,
                            OnArtWorkListFragmentInteractionListener listener,
                            Context context) {
        mArtWork = artWork;

        //TODO: The application may be doing too much work on its main thread.
        String smallImagePath = mArtWork.getSmallImagePath();
        if (smallImagePath != null) {
            Bitmap smallArtWorkImage = Gallery.get(context).getArtWorkImage(smallImagePath);
            mArtWorkImageView.setImageBitmap(smallArtWorkImage);
        }

        mArtworkTakenButton.setVisibility(artWork.isTaken() ? View.VISIBLE : View.INVISIBLE);

        mShowIdTextView.setText(artWork.getShowId().equals("x") ? "*" : "(" + artWork.getShowId() + ")");
        mTitleTextView.setText(artWork.getTitle());
        mArtistTextView.setText(artWork.getArtist());
        mMediaTextView.setText(
                mArtWork.getHeight().isEmpty() && mArtWork.getWidth().isEmpty()
                        ? mArtWork.getMedia()
                        : String.format("%s (%s\" x %s\")", mArtWork.getMedia(), mArtWork.getWidth(), mArtWork.getHeight())
        );

        if (mArtWork.getStars() > 0) {
            mArtWorkRatingStarImageView.setImageResource(R.drawable.ic_star_border_black_18dp);
            mArtWorkRatingTextView.setText(String.format(Locale.US, "%s", Integer.toString(mArtWork.getStars())));
            mArtWorkRatingStarImageView.setVisibility(View.VISIBLE);
            mArtWorkRatingTextView.setVisibility(View.VISIBLE);
        }
        else {
            mArtWorkRatingStarImageView.setVisibility(View.GONE);
            mArtWorkRatingTextView.setVisibility(View.GONE);
        }

        mListener = listener;
    }

    @Override
    public void onClick(View view) {

        if (mListener != null) {

            // Notify the active callbacks interface (the activity, if the
            // fragment is attached to one) that an item has been selected.
            mListener.onArtWorkListFragmentInteraction(mArtWork);
        }
    }
}

File: app/src/main/java/com/ryanpconnors/art_thief/rate/ArtWorkListViewHolder.kt
package com.ryanpconnors.art_thief.rate

import android.content.Context
import android.graphics.Bitmap
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView

import com.ryanpconnors.art_thief.R
import com.ryanpconnors.art_thief.artgallery.ArtWork
import com.ryanpconnors.art_thief.artgallery.Gallery
import com.ryanpconnors.art_thief.rate.ArtWorkListFragment.OnArtWorkListFragmentInteractionListener

import java.util.Locale

/**
 * Created by Ryan Connors on 2/17/16.
 */
class ArtWorkListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

    private val mArtWorkImageView: ImageView
    private val mArtworkTakenButton: Button
    private val mShowIdTextView: TextView
    private val mTitleTextView: TextView
    private val mArtistTextView: TextView
    private val mMediaTextView: TextView
    private val mArtWorkRatingStarImageView: ImageView
    private val mArtWorkRatingTextView: TextView

    private var mListener: OnArtWorkListFragmentInteractionListener? = null

    private var mArtWork: ArtWork? = null

    init {
        itemView.setOnClickListener(this)

        mArtWorkImageView = itemView.findViewById<View>(R.id.artwork_list_item_artwork_image) as ImageView
        mArtworkTakenButton = itemView.findViewById<View>(R.id.artwork_list_item_taken_button) as Button
        mShowIdTextView = itemView.findViewById<View>(R.id.artwork_list_item_show_id) as TextView
        mTitleTextView = itemView.findViewById<View>(R.id.artwork_list_item_title) as TextView
        mArtistTextView = itemView.findViewById<View>(R.id.artwork_list_item_artist) as TextView
        mMediaTextView = itemView.findViewById<View>(R.id.artwork_list_item_media) as TextView
        mArtWorkRatingStarImageView = itemView.findViewById<View>(R.id.artwork_list_item_rating_star_image) as ImageView
        mArtWorkRatingTextView = itemView.findViewById<View>(R.id.artwork_list_item_rating_text_view) as TextView
    }

    fun bindArtWork(artWork: ArtWork,
                    listener: OnArtWorkListFragmentInteractionListener,
                    context: Context) {
        mArtWork = artWork

        //TODO: The application may be doing too much work on its main thread.
        val smallImagePath = mArtWork!!.smallImagePath
        if (smallImagePath != null) {
            val smallArtWorkImage = Gallery.get(context).getArtWorkImage(smallImagePath)
            mArtWorkImageView.setImageBitmap(smallArtWorkImage)
        }

        mArtworkTakenButton.visibility = if (artWork.isTaken) View.VISIBLE else View.INVISIBLE

        mShowIdTextView.text = if (artWork.showId == "x") "*" else "(" + artWork.showId + ")"
        mTitleTextView.text = artWork.title
        mArtistTextView.text = artWork.artist
        mMediaTextView.text = if (mArtWork!!.height!!.isEmpty() && mArtWork!!.width!!.isEmpty())
            mArtWork!!.media
        else
            String.format("%s (%s\" x %s\")", mArtWork!!.media, mArtWork!!.width, mArtWork!!.height)

        if (mArtWork!!.stars > 0) {
            mArtWorkRatingStarImageView.setImageResource(R.drawable.ic_star_border_black_18dp)
            mArtWorkRatingTextView.text = String.format(Locale.US, "%s", Integer.toString(mArtWork!!.stars))
            mArtWorkRatingStarImageView.visibility = View.VISIBLE
            mArtWorkRatingTextView.visibility = View.VISIBLE
        } else {
            mArtWorkRatingStarImageView.visibility = View.GONE
            mArtWorkRatingTextView.visibility = View.GONE
        }

        mListener = listener
    }

    override fun onClick(view: View) {

        if (mListener != null) {

            // Notify the active callbacks interface (the activity, if the
            // fragment is attached to one) that an item has been selected.
            mListener!!.onArtWorkListFragmentInteraction(mArtWork)
        }
    }
}
